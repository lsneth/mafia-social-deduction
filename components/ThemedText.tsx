import { Text, type TextProps } from 'react-native'

export type ThemedTextProps = TextProps & {
  type?: 'default' | 'title' | 'subtitle'
  underline?: boolean
}

export function ThemedText({ className, type = 'default', underline = false, ...rest }: ThemedTextProps): JSX.Element {
  return (
    <Text
      style={{ fontFamily: type === 'title' ? 'Oswald_700Bold' : 'CrimsonText_400Regular' }}
      className={`text-mafiaWhite ${
        type === 'title' ? 'text-7xl pt-5' : type === 'subtitle' ? 'text-3xl' : 'text-lg'
      } ${underline ? 'underline' : ''} ${className}`}
      {...rest}
    />
  )
}
