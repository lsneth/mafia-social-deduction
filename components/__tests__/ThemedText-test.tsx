import * as React from 'react'
import renderer from 'react-test-renderer'

import { ThemedText } from '../ThemedText'

it(`renders correctly`, () => {
  const tree = renderer.create(<ThemedText>This is a unit test for the ThemedText component</ThemedText>).toJSON()

  expect(tree).toMatchSnapshot()
})
