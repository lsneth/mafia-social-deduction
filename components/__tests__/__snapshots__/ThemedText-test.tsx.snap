// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#fff",
      },
      {
        "fontSize": 18,
        "lineHeight": 28,
      },
      {
        "textAlign": "center",
      },
      {
        "fontFamily": "CrimsonText_400Regular",
      },
    ]
  }
>
  This is a unit test for the ThemedText component
</Text>
`;
